hypershift-operator/controllers/hostedcluster/apiproviders/platform_reconciler.go

```diff
@@ -0,0 +1,33 @@
+package apiproviders
```

Comment:
may be call this folder/package just "providers" or even better let's use the same term i.e "platforms" everywhere to be consistent with the term used for naming the files, API fields and variables?

Also this package should be internal/ as the only consumers are meant to be ourselves.